torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -317,10 +307,7 @@ def _get_handle_indices(
         """
```

Comment:
Before this PR (making note to myself):

https://github.com/pytorch/pytorch/blob/be4c9a87f190a55a4679799ec140c9b85da02c6a/torch/distributed/fsdp/_exec_order_utils.py#L193-L201
https://github.com/pytorch/pytorch/blob/be4c9a87f190a55a4679799ec140c9b85da02c6a/torch/distributed/fsdp/_exec_order_utils.py#L235

The above looks buggy. If `optional_local_indices` has a `None` in it, we should not be able to pass it to `torch.tensor()`.